src/components/RoadmapPageContent/Roadmap/Roadmap.tsx

```diff
@@ -0,0 +1,89 @@
+import { useState } from 'react';
+
+import { Vertex } from 'src/types';
+import { getDistanceBetweenElements, splitVertices } from 'src/utils';
+
+import RoadmapStepModal from 'components/modals/RoadmapStepModal';
+import RoadmapAdvices from 'components/RoadmapPageContent/RoadmapAdvices';
+import RoadmapStep from 'components/RoadmapPageContent/RoadmapStep';
+import Container from 'components/ui/Container';
+
+import styles from './Roadmap.module.css';
+
+interface RoadmapProps {
+  title: string;
+  roadmapSteps: Vertex[];
+}
+
+const Roadmap = ({ title, roadmapSteps }: RoadmapProps) => {
+  const [openedDropdownId, setOpenedDropdownId] = useState<number | null>(null);
+  const [modalContent, setModalContent] = useState<Vertex | null>(null);
+  const { mainStep, subSteps } = splitVertices(roadmapSteps);
+
+  const handleModalVisibilityContent = (step: Vertex) => () =>
+    setModalContent((p) => {
+      return p === null ? step : null;
+    });
+
```

Comment:
I think we more clear make 2 sort functions () =>  setModalContent and clear content